probe-rs/src/architecture/arm/core/armv7m.rs

```diff
@@ -642,13 +642,18 @@ impl<'probe> CoreInterface for Armv7m<'probe> {
         // Wait until halted state is active again.
         let start = Instant::now();
 
-        while start.elapsed() < timeout {
+        // This will always call `self.core_halted()` at least once, which ensures that the `core` status is updated.
+        loop {
```

Comment:
The `loop` statement is discouraged. Why do you think this restructure is necessary?

Reply:
With the previous `while` statement, if the timeout had already expired by the time it got to the first `start.elapsed() < timeout`,  then the very next statement is an `Err()` ... which doesn't take into consideration whether or not the chip is actually halted.

Reply:
I'd rather check that the core is halted at the end again.

Reply:
`do {} while` would certainly be nice here :D

Reply:
Is this better?

Reply:
Yep :)